arch/arm/soc/arm/mps2/soc.h

```diff
@@ -7,6 +7,12 @@
 #ifndef _SOC_H_
 #define _SOC_H_
 
+#if defined(CONFIG_SOC_MPS2_AN521)
```

Comment:
Since this SOC implements the Security Extension, I think we need to define the CMSE feature:
`#define __ARM_FEATURE_CMSE 3`
so that we are able to  use the ARM CMSE API (for security extension)

Reply:
Related to the above comment: I think we will also need the -mcmse option for gccarmemb compile options (probably in arch/arm/CMakeLists.txt).
Maybe this does not belong to this PR, though.

Reply:
Hmm, good to know.  Seems like we should have a Kconfig symbol for CMSE, do you have that as part of the security extension support you've done?

Reply:
No, and the reason is that so far I have been contributing code that does not use the CMSE API. I suppose we need something like

select ARM_CMSE inside ARM_SECURE_FIRMWARE

Reply:
Probably so - we can figure that out as we develop the support.  I'll hold off on it for now w/regards to this base level SoC/board support

Reply:
I just remember I had created an issue: https://github.com/zephyrproject-rtos/zephyr/issues/6402